python/samples/getting_started_with_processes/step03/step03a_food_preparation.py

```diff
@@ -1,126 +1,428 @@
+    """
+    Helper function that:
+     1. Builds a KernelProcess from a ProcessBuilder
+     2. Starts it with the given external event
+     3. Waits for completion
+    """
     kernel = _create_kernel_with_chat_completion("sample")
+
     kernel_process = process.build()
     print(f"=== Start SK Process '{process.name}' ===")
-    _ = await start(
-        process=kernel_process, kernel=kernel, initial_event=KernelProcessEvent(id=external_trigger_event, data=[])
-    )
-    print(f"=== End SK Process '{process.name}' ===")
-
-
-async def execute_process_with_state(process, kernel, external_trigger_event, order_label) -> KernelProcess:
+    async with await start(
+        kernel_process,
+        kernel,
+        KernelProcessEvent(id=external_trigger_event, data=[]),
+    ):
+        pass
+    print(f"=== End SK Process '{process.name}' ===\n")
+
+
+async def execute_process_with_state(
+    process: KernelProcess, kernel: Kernel, external_trigger_event: str, order_label: str
+) -> KernelProcess:
+    """
+    Starts the provided KernelProcess (with optional existing state),
+    returns the updated state after the run.
+    """
     print(f"=== {order_label} ===")
-    async with await start(process, kernel, KernelProcessEvent(id=external_trigger_event, data=[])) as running_process:
+    async with await start(
+        process,
+        kernel,
+        KernelProcessEvent(id=external_trigger_event, data=[]),
+    ) as running_process:
         return await running_process.get_state()
 
 
 # endregion
 
+
+# region Local JSON file handling for process state
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+# Build the path to "step03/processes_states"
+PROCESS_STATE_DIRECTORY = BASE_DIR / "step03" / "processes_states"
+PROCESS_STATE_DIRECTORY.mkdir(parents=True, exist_ok=True)
+
+
+def dump_process_state_metadata_locally(process_state: KernelProcessStateMetadata, json_filename: str) -> None:
+    """
+    Saves the ProcessStateMetadata to a local JSON file in step03/processes_states,
+    relative to the current script's grandparent folder.
+    """
+    file_path = PROCESS_STATE_DIRECTORY / json_filename
+    with open(file_path, "w", encoding="utf-8") as f:
+        json.dump(process_state.model_dump(exclude_none=True, by_alias=True, mode="json"), f, indent=4)
+    print(f"Process state saved to '{file_path.resolve()}'")
+
+
+def load_process_state_metadata(json_filename: str) -> KernelProcessStateMetadata | None:
+    """
+    Loads the ProcessStateMetadata from step03/processes_states if it exists.
+    Returns None if the file doesn't exist or fails to parse.
+    """
+    file_path = PROCESS_STATE_DIRECTORY / json_filename
+    if not file_path.exists():
+        print(f"No such file: '{file_path.resolve()}'")
+        return None
+
+    try:
+        with open(file_path, encoding="utf-8") as f:
+            data = json.load(f)
+        return KernelProcessStateMetadata(**data)
```

Comment:
The json serializer in pydantic is faster then json.load, so something like this would be better:
```suggestion
        return KernelProcessStateMetadata.model_validate_json(f)
```

Reply:
and maybe we should even have a `KernelProcessStateMetadata.from_file(file_path, encoding=...)`

Reply:
Good idea. Added the convenience method.